superset/config.py

```diff
@@ -1125,9 +1128,9 @@ class CeleryConfig:  # pylint: disable=too-few-public-methods
 GLOBAL_ASYNC_QUERIES_TRANSPORT = "polling"
 GLOBAL_ASYNC_QUERIES_POLLING_DELAY = 500
 
-# It's possible to add a dataset health check logic which is specific to your system.
-# It will get executed each time when user open a chart's explore view.
-DATASET_HEALTH_CHECK = None
+# A SQL dataset health check. Note if enabled the callable should be cached to aid with
+# performance.
+DATASET_HEALTH_CHECK: Optional[Callable[["SqlaTable"], str]] = None
```

Comment:
this should probably be a `BaseDatasource` not a `SqlaTable`?

Reply:
@etr2460 this functionality is only provided for the SQLA connector.

Reply:
Would be nice do add an health check callable example here, or add it to the docs

Reply:
@dpgaspar I added an example.